rfc/2022/2022-11-02_TechTeam4_Preview_Environment_Architecture.md

```diff
@@ -0,0 +1,97 @@
+
+A diagram is submitted [below](#Diagrams) as additional explanation.
+
+**MVP Feature Set:**
+1. Preview Environment MVP will create ephemeral environments that can be accessed without SOCKS
+2. Preview Environment MVP will operate on the EKS infrastructure and paradigm
+3. Preview Environment MVP will automatically create a Preview Environment when a PR is created
+4. Preview Environment MVP will utilize mock test data
+
+### Post-MVP Design: Additional Features
+For reference, there are numerous additional features planned for inclusion in Post-MVP iterations. Specifically, we will begin adding the abilities to specify custom settings like: custom repository configurations, selection of different types of test data, connectivity to lower environment endpoints for external partners. These features will be vetted and prioritized as development progresses.
+
+## Risks
+* Removing the ability for a preview environment to integrate with VA enterprise lower environments may initially reduce the quality of the experience of preview environments for some users. These risks will lessen over time as additional functionality is added into preview environments and this capability is restored.
+* If there is an overlap between the creation of review instances and the creation of preview environments there could be increased resource usage and cost while both products are supported.
+
+<!--
+List the risks of this approach
+
+* There are always risks. What are the risks of this solution?
+* These are the things people will bring up in opposition to your idea or plans. Acknowledge them.
+-->
+## Alternatives
+There were a number of approaches explored during Discovery, listed first below is the recommended approach, followed by the two other primary solutions considered. Lastly, we included a feature comparison to inform everyone on what led into our decision making process.
+
+### Custom preview environment build process: (Selected Approach)
+A custom process that is driven by events in GitHub, built by GitHub Actions, and orchestrated by EKS. The capabilities with a custom solution are limited by few characteristics of the existing tech stack. Known limitations within the technologies available for our custom solution are that GitHub API limits the number of requests we can make, EKS will only operate within certain VPCs, and creating a new subdomain for use with preview environments may be a lengthy process.
+
+### GitHub Codespaces:
+Codespaces is billed by GitHub as a "dev environment in the cloud." While Codespaces does create development environments, the primary purpose of these development environments is focused around coding and controlling the creation of reproducible development environments for the purposes of collaboration between developers and the facilitation of onboarding new developers.
```

Comment:
How does this align or not align with the goals of Preview Environments? Codespaces creates ephemeral cloud instances which can be automated, configured, and exposed to users.

Reply:
Expanded the explanation of Codespaces limitations in the RFC